Jenkinsfile

```diff
@@ -0,0 +1,599 @@
+  env.GITHUB_COMMIT_URL = "${GITHUB_PROJECT_URL}/commit/${env.GIT_COMMIT}"
+
+  env.MAVEN_HOME = "${env.HOME}/.mvn/apache-maven-3.3.9"
+  env.PATH = "${env.MAVEN_HOME}/bin:${env.PATH}"
+  env.JAVA_HOME = sh(label: 'Get JAVA_HOME',script: '''#!/bin/bash -lex
+    . ${JABBA_SHELL}
+    jabba which ${JABBA_VERSION}''', returnStdout: true).trim()
+  env.JAVA8_HOME = sh(label: 'Get JAVA8_HOME',script: '''#!/bin/bash -lex
+    . ${JABBA_SHELL}
+    jabba which 1.8''', returnStdout: true).trim()
+
+  sh label: 'Download Apache CassandraⓇ or DataStax Enterprise',script: '''#!/bin/bash -lex
+    . ${JABBA_SHELL}
+    jabba use ${JABBA_VERSION}
+    . ${CCM_ENVIRONMENT_SHELL} ${SERVER_VERSION}
+  '''
+
+  sh label: 'Display Java and environment information',script: '''#!/bin/bash -le
+    # Load CCM environment variables
+    set -o allexport
+    . ${HOME}/environment.txt
+    set +o allexport
+
+    . ${JABBA_SHELL}
+    jabba use ${JABBA_VERSION}
+
+    java -version
+    mvn -v
+    printenv | sort
+  '''
+}
+
+def buildDriver(jabbaVersion) {
+  withEnv(["BUILD_JABBA_VERSION=${jabbaVersion}"]) {
+    sh label: 'Build driver', script: '''#!/bin/bash -lex
+      . ${JABBA_SHELL}
+      jabba use ${BUILD_JABBA_VERSION}
+
+      mvn -B -V install -DskipTests -Dmaven.javadoc.skip=true
+    '''
+  }
+}
+
+def executeTests() {
+  sh label: 'Execute tests', script: '''#!/bin/bash -lex
+    # Load CCM environment variables
+    set -o allexport
+    . ${HOME}/environment.txt
+    set +o allexport
+
+    . ${JABBA_SHELL}
+    jabba use ${JABBA_VERSION}
+
+    if [ "${JABBA_VERSION}" != "1.8" ]; then
+      SKIP_JAVADOCS=true
+    else
+      SKIP_JAVADOCS=false
+    fi
+
+    INTEGRATION_TESTS_FILTER_ARGUMENT=""
+    if [ ! -z "${INTEGRATION_TESTS_FILTER}" ]; then
+      INTEGRATION_TESTS_FILTER_ARGUMENT="-Dit.test=${INTEGRATION_TESTS_FILTER}"
+    fi
+    printenv | sort
+
+    mvn -B -V ${INTEGRATION_TESTS_FILTER_ARGUMENT} verify \
+      -Dmaven.javadoc.skip=${SKIP_JAVADOCS} \
+      -Dccm.version=${CCM_CASSANDRA_VERSION} \
+      -Dccm.dse=${CCM_IS_DSE} \
+      -Dproxy.path=${HOME}/proxy \
+      ${SERIAL_ITS_ARGUMENT} \
+      ${ISOLATED_ITS_ARGUMENT} \
+      ${PARALLELIZABLE_ITS_ARGUMENT}
+  '''
+}
+
+def executeCodeCoverage() {
+  jacoco(
+    execPattern: '**/target/jacoco.exec',
+    classPattern: '**/classes',
+    sourcePattern: '**/src/main/java'
+  )
+}
+
+def notifySlack(status = 'started') {
+  // Notify Slack channel for every build except adhoc executions
+  if (params.ADHOC_BUILD_TYPE != 'BUILD-AND-EXECUTE-TESTS') {
+    // Set the global pipeline scoped environment (this is above each matrix)
+    env.BUILD_STATED_SLACK_NOTIFIED = 'true'
+
+    def buildType = 'Commit'
+    if (params.CI_SCHEDULE != 'DO-NOT-CHANGE-THIS-SELECTION') {
+      buildType = "${params.CI_SCHEDULE.toLowerCase().capitalize()}"
+    }
+
+    def color = 'good' // Green
+    if (status.equalsIgnoreCase('aborted')) {
+      color = '808080' // Grey
```

Comment:
why other colors are named and this is number?

Reply:
This is the Slack plugin for Jenkins. It only understands those 3 named colors, and any hex code color: [https://jenkins.io/doc/pipeline/steps/slack/](https://jenkins.io/doc/pipeline/steps/slack/)